photosynthesis_metrics/gs.py

```diff
@@ -0,0 +1,255 @@
+    RLT by formulas (2) and (3) from the paper. This function will be typically called
+    on the output of the gudhi persistence_intervals_in_dimension function.
+
+    Args:
+      intervals: list of intervals e.g. [[0, 1], [0, 2], [0, np.inf]].
+      alpha_max: The maximal persistence value
+      i_max: Upper bound on the value of beta_1 to compute.
+
+    Returns:
+        rlt: Array of size (i_max, ) containing desired RLT.
+    """
+
+    persistence_intervals = []
+    # If for some interval we have that it persisted up to np.inf
+    # we replace this point with alpha_max.
+    for interval in intervals:
+        if not np.isinf(interval[1]):
+            persistence_intervals.append(list(interval))
+        elif np.isinf(interval[1]):
+            persistence_intervals.append([interval[0], alpha_max])
+
+    # If there are no intervals in H1 then we always observed 0 holes.
+    if len(persistence_intervals) == 0:
+        rlt = np.zeros(i_max)
+        rlt[0] = 1.0
+        return rlt
+
+    persistence_intervals_ext = persistence_intervals + [[0, alpha_max]]
+    persistence_intervals_ext = np.array(persistence_intervals_ext)
+    persistence_intervals = np.array(persistence_intervals)
+
+    # Change in the value of beta_1 may happen only at the boundary points
+    # of the intervals
+    switch_points = np.sort(np.unique(persistence_intervals_ext.flatten()))
+    rlt = np.zeros(i_max)
+    for i in range(switch_points.shape[0] - 1):
+        midpoint = (switch_points[i] + switch_points[i + 1]) / 2
+        s = 0
+        for interval in persistence_intervals:
+            # Count how many intervals contain midpoint
+            if midpoint >= interval[0] and midpoint < interval[1]:
+                s = s + 1
+        if (s < i_max):
+            rlt[s] += (switch_points[i + 1] - switch_points[i])
+
+    return rlt / alpha_max
+
+
+def lmrk_table(W: np.ndarray, L: np.ndarray):
+    r"""Construct an input for the gudhi.WitnessComplex function.
+    Args:
+        W: Array of size w x d, containing witnesses
+        L: Array of size l x d, containing landmarks
+
+    Returns:
+        distances: 3D array of size w x l x 2. It satisfies the property that
+            distances[i, :, :] is [idx_i, dists_i], where dists_i are the sorted distances
+            from the i-th witness to each point in L and idx_i are the indices of the corresponding points
+            in L, e.g., D[i, :, :] = [[0, 0.1], [1, 0.2], [3, 0.3], [2, 0.4]]
+        max_dist: Maximal distance between W and L
+    """
+    a = cdist(W, L)
+    max_dist = np.max(a)
+    idx = np.argsort(a)
+    b = a[np.arange(np.shape(a)[0])[:, np.newaxis], idx]
+    distances = np.dstack([idx, b])
+    return distances, max_dist
+
+
+def witness(X, sample_size: int = 64, gamma: Optional[float] = None):
+    """Compute the persistence intervals for the dataset X using the witness complex.
+
+    Args:
+        X: Array of shape (N_samples, data_dim) representing the dataset.
```

Comment:
I would propose to use lower case variable names when possible to satisfy Goodle style guide